wirecard/oxid-ee
# WDPD-166: Guaranteed Invoice by Wirecard Frontend Payment Method

### This PR

* Adds Input fields for date of birth and phone
* Adds select field for saving input data
* Adds mechanism to show/hide payment methods(e.g. hide guaranteed invoice if user < 18)

### Notes

* Depends on wdpd-178
* if basket is empty it is not possible to get to the payment method select step in oxid
* Guaranteed Invoice by Wirecard payment method is only testable with country set to Austria, to be able to use Austria add it in backend (Master Settings->Countries)

### TODO

* Add check for digital goods
* Add check for currency
* Add check for shipping country
* Add check for billing country
* Handle different shipping and billing address handling

as soon as the configured values are available

* Write Tests

### How to Test

* Check if input fields are correctly shown/hidden to/from user
* Check if save selection is correctly shown/hidden to/from user
* Check if saving/ not saving works correctly
* Check if showing/hiding of payment method works correctly
* Test with logged in and guest user

### Jira Links
https://jira.parkside.at/browse/WDPD-166
